eclair-core/src/test/scala/fr/acinq/eclair/testutils/MySlf4jLogger.scala

```diff
@@ -0,0 +1,142 @@
+ */
+class MySlf4jLogger extends Actor with SLF4JLogging with RequiresMessageQueue[LoggerMessageQueueSemantics] {
+
+  private val contextName = context.system.settings.config.getString("akka.logging-context")
+
+  val loggerFactory: LoggerContext = MyContextSelector.Singleton.getLoggerContext(contextName)
+
+  val mdcThreadAttributeName = "sourceThread"
+  val mdcActorSystemAttributeName = "sourceActorSystem"
+  val mdcAkkaSourceAttributeName = "akkaSource"
+  val mdcAkkaTimestamp = "akkaTimestamp"
+  val mdcAkkaAddressAttributeName = "akkaAddress"
+
+  private def akkaAddress = context.system.asInstanceOf[ExtendedActorSystem].provider.rootPath.address.toString
+
+  def receive = {
+
+    case event@Error(cause, logSource, logClass, message) =>
+      withMdc(logSource, event) {
+        cause match {
+          case Error.NoCause | null =>
+            MyLogger(loggerFactory, logClass, logSource).error(markerIfPresent(event), if (message != null) message.toString else null)
+          case _ =>
+            MyLogger(loggerFactory, logClass, logSource).error(
+              markerIfPresent(event),
+              if (message != null) message.toString else cause.getLocalizedMessage,
+              cause)
+        }
+      }
+
+    case event@Warning(logSource, logClass, message) =>
+      withMdc(logSource, event) {
+        event match {
+          case e: LogEventWithCause =>
+            MyLogger(loggerFactory, logClass, logSource).warn(
+              markerIfPresent(event),
+              if (message != null) message.toString else e.cause.getLocalizedMessage,
+              e.cause)
+          case _ =>
+            MyLogger(loggerFactory, logClass, logSource).warn(markerIfPresent(event), if (message != null) message.toString else null)
+        }
+      }
+
+    case event@Info(logSource, logClass, message) =>
+      withMdc(logSource, event) {
+        MyLogger(loggerFactory, logClass, logSource).info(markerIfPresent(event), "{}", message: Any)
+      }
+
+    case event@Debug(logSource, logClass, message) =>
+      withMdc(logSource, event) {
+        MyLogger(loggerFactory, logClass, logSource).debug(markerIfPresent(event), "{}", message: Any)
+      }
+
+    case InitializeLogger(_) =>
+      log.info("Slf4jLogger started")
+      sender() ! LoggerInitialized
+  }
+
+  @inline
+  final def withMdc(logSource: String, logEvent: LogEvent)(logStatement: => Unit): Unit = {
+    logEvent match {
+      case m: LogEventWithMarker if m.marker ne null =>
+        val properties = m.marker.properties
+        if (properties.nonEmpty) {
+          properties.foreach { case (k, v) => MDC.put(k, String.valueOf(v)) }
+        }
+      case _ =>
+    }
+
+    MDC.put(mdcAkkaSourceAttributeName, logSource)
+    MDC.put(mdcThreadAttributeName, logEvent.thread.getName)
+    MDC.put(mdcAkkaTimestamp, formatTimestamp(logEvent.timestamp))
+    MDC.put(mdcActorSystemAttributeName, context.system.name)
+    MDC.put(mdcAkkaAddressAttributeName, akkaAddress)
```

Comment:
Do we really care about those?

Reply:
Not really, it's just to keep the diff minimal